src/presign/participant.rs

```diff
@@ -253,27 +266,32 @@ impl PresignParticipant {
             let (pr, mut r2_msg) = self.handle_round_one_msg(rng, &msg, main_storage)?;
             out_messages.append(&mut r2_msg);
 
-            // Check that pr is only ever assigned to at most once.
+            // Check that `presign_record` is only ever assigned to at most once.
             match (pr, &presign_record) {
                 // Found some _pr_ and presign_record has never been assigned to. Assign to it.
                 (Some(pr), None) => presign_record = Some(pr),
                 // We have already assigned to presign_record once! This should not happen again!
                 // TODO: Add logging message here once we have logging set up.
-                (Some(_), Some(_)) => return Err(InternalInvariantFailed),
+                (Some(_), Some(_)) => {
+                    error!("`presign_record` has already been assigned to once.");
+                    return Err(InternalInvariantFailed);
+                }
                 (None, _) => { /* Nothing to do */ }
             }
         }
 
         Ok((presign_record, out_messages))
     }
 
+    #[instrument(skip_all, err(Debug))]
     fn handle_round_one_broadcast_msg<R: RngCore + CryptoRng>(
         &mut self,
         rng: &mut R,
         broadcast_message: &BroadcastOutput,
         main_storage: &Storage,
     ) -> Result<(Option<PresignRecord>, Vec<Message>)> {
         if broadcast_message.tag != BroadcastTag::PresignR1Ciphertexts {
+            error!("Incorrect tag for Presign R1 Broadcast!");
```

Comment:
Is this not already captured by the `bail!` macro? If not, should we add this `error!` call there instead of duplicating the text here?

Reply:
I just rebased Tom's PR which gets rid of the `bail!` macro usage. So we no longer duplicate the text.